serverless/cmd/feeder/impl/feeder.go

```diff
@@ -172,14 +173,18 @@ func submitToWitness(ctx context.Context, cpRaw []byte, cpSubmit log.Checkpoint,
 
 		var conP [][]byte
 		if len(latestCPRaw) > 0 {
-			_, err := note.Open(latestCPRaw, note.VerifierList(logSigV))
+			n, err := note.Open(latestCPRaw, note.VerifierList(logSigV, wSigV))
 			if err != nil {
 				glog.Warningf("%s: failed to open CP: %v", wSigV.Name(), err)
 				continue
 			}
+			if numSigs := len(n.Sigs); numSigs != 2 {
+				errs <- errors.New("checkpoint from witness was not signed by at least log + witness")
```

Comment:
The condition requires exactly 2, but the message says "at least [2]".

Reply:
The CP could've been signed by Log, Witness, and some number of unknown other witnesses.
The _known sigs_ are max(2) 'cos we only provide the log and witness verifiers, though.